### ISSUE TITLE
:lady_beetle: tar -I lrzip-next -x|t will fail (Compression works fine)

### ISSUE THREAD

**Issue by pete4abw** [pete4abw/lrzip-next#81]
### lrzip-next Version

0.9.0

### lrzip-next command line

`tar -I lrzip-next -tvvf file.tar.lrz`

### What happened?

`tar -I lrzip-next` fails when extracting or testing (listing) files returns an error due to faulty pipeing.

```
$ tar -I lrzip-next -tvvf lrzip-next.tar.lrz
Using configuration file /home/peter/.lrzip/lrzip.conf
The following options are in effect for this DECOMPRESSION.
Threading is ENABLED. Number of CPUs detected: 8
Detected 16,558,071,808 bytes ram
Nice Value: 19
Show Progress
Verbose
Temporary Directory set as: ./
Unknown hash, falling back to CRC
Outputting to stdout.
Detected lrzip version 0.9 file.
(null) being used for integrity testing.
tar: Child died with signal 11
tar: Error is not recoverable: exiting now
```

### What was expected behavior?

File extraction or listing would succeed

### Steps to reproduce

Execute command as above.

### Please provide system details

OS Distro: Slackware
Kernel Version (uname -a):  5.18.10
System ram (free -h): 
```
$ free -h
               total        used        free      shared  buff/cache   available
Mem:            15Gi       1.3Gi        12Gi       421Mi       2.0Gi        13Gi
Swap:           15Gi          0B        15Gi
```

### Additional Context

Note to self: review `lrzip.c` in the `decompress_file` function to see why TEST_ONLY and STDOUT are missing the mark.

ITMT, output `lrzip-next` to tar like this this:

```
$ lrzip-next -d -o - lrzip-next.tar.lrz | tar -tvvf -
Using configuration file /home/peter/.lrzip/lrzip.conf
The following options are in effect for this DECOMPRESSION.
Threading is ENABLED. Number of CPUs detected: 8
Detected 16,558,071,808 bytes ram
Nice Value: 19
Show Progress
Verbose
Output Filename Specified: -
Temporary Directory set as: ./
Outputting to stdout.
Detected lrzip version 0.9 file.
SHA256 being used for integrity testing.
Validating file for consistency...[OK]
Detected lrzip version 0.9 file.
drwxr-xr-x peter/users       0 2022-06-22 08:17 lrzip-next/
-rw-r--r-- peter/users    4112 2021-03-05 06:31 lrzip-next/README-NOT-BACKWARD-COMPATIBLE
-rw-r--r-- peter/users     431 2021-03-05 06:31 lrzip-next/.gitignore
drwxr-xr-x peter/users       0 2022-06-21 11:16 lrzip-next/man/
-rw-r--r-- peter/users    5010 2022-06-21 09:03 lrzip-next/man/lrznunzip.1
```

**Comment by pete4abw** [comment]
A review points to the function where comments are fetched. This bug was introduced with the expansion of the magic header. `lrzip-next` functions normally except when extracting using `tar -I lrzip-next`. More testing will be done. I'll get to it when I can.

**Comment by pete4abw** [comment]
fixed in [v0.9.1](https://github.com/pete4abw/lrzip-next/tree/v0.9.1). See **lzma-22.00** and **whats-next** branches.
